perform.pl

```diff
@@ -97,17 +113,42 @@ sub good {
 my $remote_ssh = $ENV{REMOTE_SSH}
     or die "Environemnt REMOTE_SSH not set";
 
+if ($testmode{iperftcp6} || $testmode{iperfudp6} || $testmode{net6} ||
+	$testmode{tcp6} || $testmode{udp6} || $testmode{iperf6} ||
+	$testmode{tcpbench6} || $testmode{udpbench6}) {
+    my $local_addr6 = $ENV{LOCAL_ADDR6}
+	or die "Environemnt LOCAL_ADDR6 not set";
+    my $remote_addr6 = $ENV{REMOTE_ADDR6}
+	or die "Environemnt REMOTE_ADDR6 not set";
+}
+
 # iperf3 and tcpbench tests
 
-if ($testmode{iperftcp} || $testmode{iperfudp}) {
+if ($testmode{iperftcp4} || $testmode{iperfudp4}) {
     my @sshcmd = ('ssh', $remote_ssh, 'pkill', 'iperf3');
     system(@sshcmd);
     @sshcmd = ('ssh', '-f', $remote_ssh, 'iperf3', '-s', '-D');
     system(@sshcmd)
 	and die "Start iperf3 server with '@sshcmd' failed: $?";
 }
 
-if ($testmode{tcpbench}) {
+if ($testmode{iperftcp6} || $testmode{iperfudp6}) {
+    my @sshcmd = ('ssh', $remote_ssh, 'pkill', 'iperf3');
+    system(@sshcmd);
+    @sshcmd = ('ssh', '-f', $remote_ssh, 'iperf3', '-s', '-D');
```

Comment:
You should force iperf3, tcpbench, udpbench to IPv6 mode.  Otherwise these IPv4 tests run twice.

Reply:
I also added -4 for the other case.

Reply:
should there be a remote_ssh6?